test/unit/bindingpatterns.spec.ts

```diff
@@ -90,3 +90,21 @@ test.each(testCases)(
         expect(result).toBe(true);
     },
 );
+
+test.each([
+    { bindingString: "{x, y = true}", objectString: "{x: false, y: false}", returnVariable: "y" },
+    {
+        bindingString: "{x, y: [z = true]}",
+        objectString: "{x: false, y: [false]}",
+        returnVariable: "z",
+    },
```

Comment:
Could you add a test with simply the pattern `[x = false]` too?

Reply:
Array binding patterns currently can't have default parameters. I'll add this with another commit, doesn't look like it resulted in much more code to review